.github/workflows/ci.yml

```
@@ -0,0 +1,46 @@
+on:
+  push:
+  pull_request:
+
+jobs:
+  generate-winmd:
+    name: Generate winmd
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Clean
+        run: Remove-Item -Recurse -Force .windows/winmd/
+      - uses: ilammy/msvc-dev-cmd@v1 # Needed to find CL.EXE
+      - name: Generate
+        run:  dotnet build
+        working-directory: .metadata/
+      - name: Upload
+        uses: actions/upload-artifact@v3
+        with:
+          name: metadata
+          # TODO: Only need .windows/winmd/Microsoft.States.winmd,
+          # the core Win32 bindings are 15MB and don't seem needed
+          # to generate the Rust code since windows-metadata already
+          # includes these files.
+          path: .windows/winmd
+
+  generate-rust:
+    name: Generate Rust crate
+    runs-on: ubuntu-latest # See if this runs on Linux
+    needs: generate-winmd
+    steps:
+      - uses: actions/checkout@v3
+      - name: Clean
+        run: rm -rf .windows/winmd/ src/Microsoft/
+      - name: Download
+        uses: actions/download-artifact@v3
+        with:
+          name: metadata
+          path: .windows/winmd
+      - name: Generate
+        run: cargo r -p tools_api
```

I always prefer to be explicit

```suggestion
        run: cargo run -p tools_api
```